generate-manifest

```diff
@@ -0,0 +1,157 @@
+    run_command([
+        'flatpak', 'remote-add', '--if-not-exists',
+        'flathub', 'https://flathub.org/repo/flathub.flatpakrepo'
+    ])
+    run_command([
+        'flatpak', 'install', '--noninteractive', '-y',
+        '--or-update', 'flathub', get_kolibri_flatpak_ref()
+    ])
+
+
+def run_kolibri(args):
+    run_command([
+        'flatpak', 'run', '--command=/app/libexec/kolibri',
+        '--filesystem={}'.format(BASE_PATH), get_kolibri_flatpak_ref(),
+    ] + args, env={
+        'KOLIBRI_HOME': KOLIBRI_HOME_PATH,
+    })
+
+
+def import_channel(channel_spec_str):
+    channel_spec = urllib.parse.parse_qs(channel_spec_str)
+
+    channel_id = channel_spec.get('channel_id', [None])[0]
+    node_ids = channel_spec.get('node_ids', [None])[0]
+    exclude_node_ids = channel_spec.get('exclude_node_ids', [None])[0]
+
+    node_ids = [] if node_ids is None else node_ids.split(',')
+    exclude_node_ids = [] if exclude_node_ids is None else exclude_node_ids.split(',')
+
+    if not channel_id:
+        raise ValueError('channel_id is not valid')
+
+    run_kolibri([
+        'manage', 'importchannel', 'network', channel_id,
+    ])
+
+    if node_ids:
+        run_kolibri(
+            ['manage', 'importcontent'] +
+            (['--node_ids', ','.join(node_ids)] if node_ids else []) +
+            (['--exclude_node_ids', ','.join(exclude_node_ids)] if exclude_node_ids else []) +
+            ['network', channel_id]
+        )
+
+    return {
+        'channel_id': channel_id,
+        'node_ids': node_ids,
+        'exclude_node_ids': exclude_node_ids,
+    }
+
+
+def get_output_filename(template_filename):
+    return os.path.splitext(template_filename)[0]
+
+
+def replace_placeholders(content, **kwargs):
+    args = {
+        '@ID@': kwargs['id'],
+        '@BRANCH@': kwargs['branch'],
+    }
+    for key in args.keys():
+        content = content.replace(key, args[key])
+    return content
+
+
+def generate_manifest(**kwargs):
+    with open(MANIFEST_TEMPLATE_PATH, 'r') as ftemplate:
+        with open(get_output_filename(MANIFEST_TEMPLATE_PATH), 'w') as foutput:
+            foutput.write(replace_placeholders(ftemplate.read(), **kwargs))
+
+
+def get_kolibri_flatpak_ref():
+    if _KOLIBRI_FLATPAK_REF is not None:
+        return _KOLIBRI_FLATPAK_REF
+
+    with open(get_output_filename(MANIFEST_TEMPLATE_PATH), 'r') as f:
+        manifest = json.load(f)
+    return 'app/{}//{}'.format(manifest['runtime'], manifest['runtime-version'])
+
+
+def main():
+    ap = argparse.ArgumentParser(
+        allow_abbrev=False,
+        description='Generate org.learningequality.Kolibri.Content '
+                    'Flatpak manifest for selected content'
+    )
+    ap.add_argument('--id', required=True,
+                    help='The ID of the package, it is used as the extension '
+                         'Flatpak ID last segment')
+    ap.add_argument('--branch', default='1.0',
+                    help='The Flatpak branch')
```

Comment:
This is supposed to match the branch specified in the extension point on the main Flatpak, right? So perhaps:

```suggestion
                    help='The branch specified in the extension point definition')
```

(or better wording)